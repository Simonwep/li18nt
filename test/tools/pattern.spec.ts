import {pattern} from '@lib';
import {describe, expect, it} from 'vitest';

describe('Test pattern tool', () => {

    it('Should do nothing without options', () => {
        expect(pattern({
            'home': {
                'back': 'Back',
                'welcome': {
                    'footer': 'Footer',
                    'header': 'Header'
                }
            }
        })).toMatchSnapshot();
    });

    it('Should validate a simple object', () => {
        expect(pattern({
            'abOut': 'About',
            'home': {
                'back': 'Back',
                '1next': 'Next',
                'welcome': {
                    'footer': 'Footer',
                    'heaDer': 'Header'
                }
            }
        }, {
            patterns: [
                /^[a-z]+$/
            ]
        })).toMatchSnapshot();
    });
});
