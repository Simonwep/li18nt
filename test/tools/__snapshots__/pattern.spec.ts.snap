// Vitest Snapshot v1

exports[`Test pattern tool > Should do nothing without options 1`] = `[]`;

exports[`Test pattern tool > Should validate a simple object 1`] = `
[
  {
    "failed": [
      "/^[a-z]+$/",
    ],
    "key": "abOut",
    "path": [
      "abOut",
    ],
  },
  {
    "failed": [
      "/^[a-z]+$/",
    ],
    "key": "1next",
    "path": [
      "home",
      "1next",
    ],
  },
  {
    "failed": [
      "/^[a-z]+$/",
    ],
    "key": "heaDer",
    "path": [
      "home",
      "welcome",
      "heaDer",
    ],
  },
]
`;
